dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,541 @@
+        Increasing ``tol`` will tend to reduce training time, at the cost
+        of worse models.
+
+    **kwargs : dict, optional
+        Parameters to pass to
+        :func:`~dask_ml.model_selection.IncrementalSearchCV`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from dask_ml.model_selection import HyperbandSearchCV
+    >>> from dask_ml.datasets import make_classification
+    >>> from sklearn.linear_model import SGDClassifier
+    >>>
+    >>> X, y = make_classification(chunks=20)
+    >>> est = SGDClassifier(tol=1e-3)
+    >>> params = {'alpha': np.logspace(-4, 0, num=1000),
+    >>>           'loss': ['hinge', 'log', 'modified_huber', 'squared_hinge'],
+    >>>           'average': [True, False]}
+    >>>
+    >>> search = HyperbandSearchCV(est, params)
+    >>> search.fit(X, y, classes=np.unique(y))
+    >>> search.best_params_
+    {'loss': 'log', 'average': False, 'alpha': 0.0080502}
+
+    """
+    + INC_ATTRS
+    + """
+    Notes
+    -----
+    In ``model_id``, the bracket ID prefix corresponds to how strongly that
+    bracket adapts to history. i.e., ``bracket=0`` corresponds to a completely
+    passive bracket that doesn't adapt at all.
+
+    Setting Hyperband parameters
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    To set ``max_iter`` and the chunk size for ``X`` and ``y``, it is required
+    to know
+
+    * how many "epochs" or "passes through ``X``" to train the model for
+      (``epochs`` below)
+    * a rough idea of how many hyper-parameter combinations to sample (``params_to_sample`` below)
+
+    To determine the chunk size and ``max_iter``,
+
+    1. Let ``max_iter = params_to_sample``
+    2. Let the chunks size be ``chunks_size = epochs * len(X) / params_to_sample``
+
+    Then, the estimator that sees the most examples see
+    ``max_iter * chunks_size = len(X) * epochs`` examples. Hyperband will
+    actually sample some more hyper-parameter combinations, so a rough idea of
+    parameters to sample works.
+
+    If the search space is complex, evaluate more estimators initially. Increase
+    ``params_to_sample`` by a factor of 2, and decrease ``chunk_size`` by a
+    factor of 2.
+
+    Limitations
+    ^^^^^^^^^^^
+    There are some limitations to the `current` implementation of Hyperband:
+
+    1. The full dataset is requested to be in distributed memory
+    2. The testing dataset should fit in the memory of a single worker
+
+    References
+    ----------
+    .. [1] "Hyperband: A novel bandit-based approach to hyperparameter
+           optimization", 2016 by L. Li, K. Jamieson, G. DeSalvo, A.
+           Rostamizadeh, and A. Talwalkar.  https://arxiv.org/abs/1603.06560
+
+    """
+)
+
+
+class HyperbandSearchCV(IncrementalSearchCV):
+    __doc__ = DOC
+
+    def __init__(
+        self,
+        model,
+        params,
+        max_iter,
+        aggressiveness=3,
+        patience=False,
+        tol=1e-3,
+        **kwargs
```

Comment:
Can't have `**kwargs` in the init. We'll need to copy over those parameters that aren't already here.

Reply:
Ideally the function signature would be a superset of IncrementalSearchCV, with same order for all the common arguments.